internal/service/handlers/verify_passport.go

```diff
@@ -292,12 +270,12 @@ func fulfillOrClaimPassportScanEvent(r *http.Request, balance data.Balance) erro
 	if err != nil {
 		return fmt.Errorf("get open passport scan event: %w", err)
 	}
-
 	if event == nil {
 		return errors.New("inconsistent state: balance is not verified, event type is active, but no open event was found")
 	}
 
-	if !evTypePassport.AutoClaim {
+	if !evTypePassport.AutoClaim || balance.ReferredBy == nil {
```

Comment:
If you follow this order:
1. Create balance without referral code
2. Verify passport
3. Activate balance

Passport scan event will be fulfilled, not claimed.

Suggestion: filtering event by status data.EventOpen and data.EventFulfilled

Reply:
good catch